### PR TITLE
Fix tests and write new

### PR DESCRIPTION
**Pull request by vladfreel**

### CODE REVIEW

File: app/controllers/employees_controller.rb [hunk 1]
```diff
@@ -1,5 +1,4 @@
-# frozen_string_literal: true
-
+# Employyes Controller
 class EmployeesController < ApplicationController
   before_action :find_employee, only: %i[show edit update destroy]
```
Comment (Tivgres) [inline app/controllers/employees_controller.rb]: ```exept %i[index new]```

File: app/controllers/projects_controller.rb [hunk 2]
```diff
@@ -1,9 +1,8 @@
-# frozen_string_literal: true
-
+# Projects Controller
 class ProjectsController < ApplicationController
 
   def create
-    if Project.create project_params
+    if Project.create! project_params
```
Comment (Tivgres) [inline app/controllers/projects_controller.rb]: Delete please ```!```

File: app/controllers/skills_controller.rb [hunk 3]
```diff
@@ -1,3 +1,4 @@
+# Skills Controller
 class SkillsController < ApplicationController
   before_action :find_skill, only: %i[edit update destroy]
```
Comment (Tivgres) [inline app/controllers/skills_controller.rb]: ```exept %i[new create]```

File: spec/controllers/projects_controller_spec.rb [hunk 4]
```diff
@@ -11,28 +11,30 @@
...
       expect do
-        post :create, params: { project: { name: 'qweqwe', employee_id: Employee.last.id } }
+        post :create, params: { project: { name: 'qweqwe', employee_id: Employee.last.id }}
       end.to(change { Project.count }.by(1)) &&
-        redirect_to(Employee.last &&
-        have_http_status(302)) &&
-        render_template('show') &&
-        eq(Employee.last)
+          redirect_to(Employee.last &&
+                          have_http_status(302)) &&
```
Comment (Tivgres) [inline spec/controllers/projects_controller_spec.rb]: wtf? Prefer to use one style. With tabs or without.

File: spec/controllers/projects_controller_spec.rb [hunk 5]
```diff
@@ -11,28 +11,30 @@
...
+      end.to(change { Project.count }.by(1)) &&
+          (change { Skill.count }.by(2))   &&
+          (change { Image.count }.by(1))
     end
   end
 
   describe 'PATCH #update' do
     it 'returns a success response' do
-      patch :update, params: { id: Project.last.id, project: { employee_id: Employee.last.id, name: 'qweqwe', description: 'qweqwe' } }
+      patch :update, params: { id: Project.last.id, project: { employee_id: Employee.last.id, name: 'qweqwe', description: 'qweqwe' }, employee_id: Employee.last.id  }
```
Comment (Tivgres) [inline spec/controllers/projects_controller_spec.rb]: Do u need now this additional field?

File: spec/controllers/projects_controller_spec.rb [hunk 6]
```diff
@@ -44,10 +46,10 @@
       expect do
         post :destroy, params: { id: Project.last.id, employee_id: Employee.last.id }
       end.to(change { Project.count }.by(-1)) &&
-        redirect_to(Employee.last &&
-        have_http_status(302)) &&
-        render_template('show') &&
-        eq(Employee.last)
+          redirect_to(Employee.last &&
+                          have_http_status(302)) &&
```
Comment (Tivgres) [inline spec/controllers/projects_controller_spec.rb]: The same about style. With or without tabs.

File: spec/controllers/employees_controller_spec.rb [hunk 7]
```diff
@@ -144,4 +192,23 @@
...
+      ResourceSkill.create(skill_id: skill_3.id,
+                           employee_id: employee_2.id,
+                           experience: 'good')
+    end
+    it 'returns a success response' do
+      expect do
+        patch :skill_experience,
+              params: {employee_id: employee_2.id,
+                       skill_experience:{ id: ResourceSkill.last.id,
+                                          experience: 'better' } }
```
Comment (drkmen) [inline spec/controllers/employees_controller_spec.rb]:
```ruby
params: {
 employee_id: employee_2.id, 
 skill_experience: { 
   id: ResourceSkill.last.id, 
   experience: 'better' 
  } 
}
```

File: spec/controllers/projects_controller_spec.rb [hunk 8]
```diff
@@ -11,28 +11,30 @@
...
-        post :create, params: { project: { name: nil, employee_id: Employee.last.id } }
-      end.to(change { Project.count }.by(0)) &&
-        redirect_to(Employee.last &&
-        have_http_status(302)) &&
-        render_template('show') &&
-        eq(Employee.last)
+        post :create, params: { project: { name: 'qweqwe', employee_id: Employee.last.id },
+                                image: {image: 'image1'},
+                                skills: {skill: {name: 'skill1'},
+                                         skill: {name: 'skill2'}}}
```
Comment (drkmen) [inline spec/controllers/projects_controller_spec.rb]: learn how is good hash\json view

File: spec/models/employee_spec.rb [hunk 9]
```diff
@@ -37,6 +58,20 @@
       expect(employee_deleted.deleted).to be false
     end
 
+    it '#avatar method' do
+      expect(employee.avatar).to eq 'user.png'
+    end
```
Comment (drkmen) [inline spec/models/employee_spec.rb]: is this the only case that method can return? I think if there is an image - image will be returned. Cover it.

File: spec/models/employee_spec.rb [hunk 10]
```diff
@@ -37,6 +58,20 @@
...
+
+    it '#friendly_name method' do
+      expect(employee.friendly_name).to eq 'wick'
+    end
+
+    it '#filter method by first_name' do
+      ResourceSkill.create(employee_id: employee.id, skill_id: skill.id)
+      result = Employee.filter_skills(Employee.all,skill.name)
```
Comment (drkmen) [inline spec/models/employee_spec.rb]: add space between args

File: spec/models/skill_spec.rb [hunk 11]
```diff
@@ -1,38 +1,80 @@
...
+  end
+  it 'should validate presence of name' do
+    record = Skill.new
+    record.name = '' # invalid state
+    record.valid? # run validations
+    expect(record.errors[:name]).to include("can't be blank")
+
+    record.name = 'foobar' # valid state
+    record.valid? # run validations
+    expect(record.errors[:name]).not_to include("can't be blank")
```
Comment (drkmen) [inline spec/models/skill_spec.rb]:
```ruby
expect(record).to be invalid
```

File: spec/models/skill_spec.rb [hunk 12]
```diff
@@ -1,38 +1,80 @@
...
+  end
+
+  context 'has_many specs' do
```
Comment (drkmen) [inline spec/models/skill_spec.rb]: has many specs?

File: app/controllers/skills_controller.rb (hunk 3 above)
Comment (polyakovigor) [reply app/controllers/skills_controller.rb]: Same

File: app/controllers/employees_controller.rb [hunk 13]
```diff
@@ -1,5 +1,4 @@
-# frozen_string_literal: true
-
+# Employyes Controller
```
Comment (polyakovigor) [inline app/controllers/employees_controller.rb]: Why did you add top-level class documentation suggested by rubocop and remove `frozen_string_literal: true` which is also suggested by rubocop? Make your decision - follow rubocop suggestions or not.

File: spec/controllers/employees_controller_spec.rb [hunk 14]
```diff
@@ -91,21 +112,70 @@
   describe 'POST #create' do
     it 'returns a success response' do
       expect do
-        post :create, params: {
-            employee: { first_name: 'qwe-qwe', last_name: 'qwe-qwe', email: 'email@example.com', role: 'programmer', password: '123123' }
-        }
+        post :create,
+             params: {
+                 employee: {
```
Comment (drkmen) [inline spec/controllers/employees_controller_spec.rb]: use 2 spaces instead of 4

File: spec/models/employee_spec.rb [hunk 15]
```diff
@@ -1,14 +1,34 @@
...
+  it 'has_one image' do
+    image1 = employee.create_image(image: '1234')
+    expect(employee.reload.image).to eq(image1)
+  end
+
+  it 'has_many resource_skills' do
+    resource_skill1 = employee.resource_skills.create!(skill_id: skill.id)
+    resource_skill2 = employee.resource_skills.create!(skill_id: skill.id)
+    expect(employee.reload.resource_skills).to eq([resource_skill1, resource_skill2])
+  end
```
Comment (drkmen) [inline spec/models/employee_spec.rb]: we don't need to test intermediate table

File: spec/models/employee_spec.rb [hunk 16]
```diff
@@ -37,6 +57,25 @@
       expect(employee_deleted.deleted).to be false
     end
 
+    it '#avatar method with default' do
+      expect(employee.avatar).to eq 'user.png'
+    end
+
+    it '#avatar method' do
+      employee.image = Image.new(image: Rails.root.join("app/assets/images/bw-test-user.png").open)
+      expect(employee.avatar).not_to eq 'user.png'
```
Comment (drkmen) [inline spec/models/employee_spec.rb]: `not_to eq 'user.png'` is wrong. You just create image, why don't you used it?
